File: typescript/test_performance_primes.ts
/**
 * Ganesh Bhat
 * https://github.com/ganeshkbhat/fastprimecalculations
 * Fastest method of Prime Number Calculations
 * Tests for Prime Numbers
*/

import { prime } from './primes';
import { isPrimeConventionalWay, isPrimeSquarerootWay, isPrimeSquarerootWayTwo } from './primes_alternate_ways';
import { performance } from 'perf_hooks';


let iterations = 10000000;

function calculateAverage(array: number[]): number {
    var total: number = 0;
    var count: number = 0;
    array.forEach(function (item: number, index: number) {
        total += item;
        count++;
    });
    return total / count;
}


function tests_performance_isPrimeConventionalWayArr(iterations: number): number {
    let isPrimeConventionalWaySum: number = 0;
    for (let i: number = 1; i <= iterations; i++) {
        let start: number = performance.now();
        isPrimeConventionalWay(30000239);
        let end: number = performance.now();
        isPrimeConventionalWaySum += (end - start);
    }
    return isPrimeConventionalWaySum;
}
console.log("isPrimeConventionalWay: ", tests_performance_isPrimeConventionalWayArr(iterations));


function tests_performance_isPrimeSquarerootWayArr(iterations: number): number {
    let isPrimeSquarerootWaySum: number = 0;
    for (let i: number = 1; i <= iterations; i++) {
        let start: number = performance.now();
        isPrimeSquarerootWay(30000239);
        let end: number = performance.now();
        isPrimeSquarerootWaySum += (end - start);
    }
    return isPrimeSquarerootWaySum;
}
console.log("isPrimeSquarerootWay: ", tests_performance_isPrimeSquarerootWayArr(iterations));


function tests_performance_primeArr(iterations: number): number {
    let primeSum: number = 0;
    for (let i: number = 1; i <= iterations; i++) {
        let start: number = performance.now();
        prime(30000239);
        let end: number = performance.now();
        primeSum += (end - start);
    }
    return primeSum;
}
console.log("prime (SUGGESTED): ", tests_performance_primeArr(iterations));


function tests_performance_isPrimeSquarerootWayTwoArr(iterations: number): number {
    let isPrimeSquarerootWayTwoSum: number = 0;
    for (let i: number = 1; i <= iterations; i++) {
        let start: number = performance.now();
        isPrimeSquarerootWayTwo(30000239);
        let end: number = performance.now();
        isPrimeSquarerootWayTwoSum += (end - start);
    }
    return isPrimeSquarerootWayTwoSum;
}
console.log("isPrimeSquarerootWayTwo: ", tests_performance_isPrimeSquarerootWayTwoArr(iterations));

File: java/test_performance_primes.java
/**
 * Ganesh Bhat
 * https://github.com/ganeshkbhat/fastprimecalculations
 * Fastest method of Prime Number Calculations
 * Tests for Prime Numbers
 * 
 * TODO - INCOMPLETE
 * 
*/

import java.lang.*;

public class TestPerformancePrimes {

    public static double calculateAverage(long[] primeArray) {
        long total = 0;
        int count = 0;
        for (long element: primeArray) {
            total += element;
            count++;
        }
        return total / count;
    }
    
    public static double testPrimesConventionalWay(long iterations)  {
        long isPrimeConventionalWaySum = 0;
        for (int i = 1; i <= iterations; i++){
            long start = System.nanoTime();
            PrimesAlternateWays.conventionalWay(30000239);
            long end = System.nanoTime();
            isPrimeConventionalWaySum += (end - start);
        }
        return isPrimeConventionalWaySum;
    }

    public static double testPrimesSquarerootWay(long iterations)  {
        long isPrimeSquarerootWaySum = 0;
        for (int i = 1; i <= iterations; i++){
            long start = System.nanoTime();
            PrimesAlternateWays.squarerootWay(30000239);
            long end = System.nanoTime();
            isPrimeSquarerootWaySum += (end - start);
        }
        return isPrimeSquarerootWaySum;
    }

    public static double testPrimesSquarerootWayTwo(long iterations)  {
        long isPrimeSquarerootWayTwoSum = 0;
        for (int i = 1; i <= iterations; i++){
            long start = System.nanoTime();
            PrimesAlternateWays.squarerootWayTwo(30000239);
            long end = System.nanoTime();
            isPrimeSquarerootWayTwoSum += (end - start);
        }
        return isPrimeSquarerootWayTwoSum;
    }

    public static double testPrimeAKSWay(long iterations)  {
        long isPrimeAKSWaySum = 0;
        for (int i = 1; i <= iterations; i++){
            long start = System.nanoTime();
            PrimesAlternateWays.primesAKSWay(30000239);
            long end = System.nanoTime();
            isPrimeAKSWaySum += (end - start);
        }
        return isPrimeAKSWaySum;
    }

    public static double testPrimes(long iterations) {
        long isPrimeSum = 0;
        for (int i = 1; i <= iterations; i++){
            long start = System.nanoTime();
            Primes.primes(30000239);
            long end = System.nanoTime();
            isPrimeSum += (end - start);
        }
        return isPrimeSum;
    }

    public static void main(String args[]) {
        
        long iterations = 10000000;

        System.out.println( "isPrimeConventionalWay Average : " + TestPerformancePrimes.testPrimesConventionalWay(iterations) );
        System.out.println( "isPrimeSquarerootWay Average : " + TestPerformancePrimes.testPrimesSquarerootWay(iterations) );
        System.out.println( "isPrimeSquarerootWayTwo Average : " + TestPerformancePrimes.testPrimesSquarerootWayTwo(iterations) );
        System.out.println( "isPrimeAKSWay Average : " + TestPerformancePrimes.testPrimeAKSWay(iterations) );
        System.out.println( "isPrime Average : " + TestPerformancePrimes.testPrimes(iterations) );

    }
}
